lib/hutch/config.rb

```diff
@@ -189,14 +190,15 @@ def self.convert_value(attr, value)
       end
     end
 
-    def self.method_missing(method, *args, &block)
-      attr = method.to_s.sub(/=$/, '').to_sym
-      return super unless default_config.key?(attr)
+    def self.define_methods
+      @config.keys.each do |key|
+        define_singleton_method(key) do
+          get(key)
+        end
 
-      if method =~ /=$/
-        set(attr, args.first)
-      else
-        get(attr)
+        define_singleton_method("#{key}=") do |val|
+          set(key, val)
+        end
```

Comment:
This is a _great_ part of this change.

With this, we can wrap the built-in `define_singleton_method(key)` into a YARD-documentable meta-programming method – which creates these `attr_accessor`-style methods using the above code.

[The YARD author uses `@macro` to great effect, in this example](https://groups.google.com/forum/#!topic/yardoc/HIUoeATOjWA).

That allows us to simplify the YARD "settings" extension to only cover collecting the `string_setting` and friends.

Question: Does this version of `.define_methods` want to allow the `params` given to `.initialize` to influence which get/set convenience methods are available?